graphql/util.go

```diff
@@ -427,46 +427,58 @@ func mapHTTPStatusToGqlError(ctx context.Context, httpStatus int, err error) *gq
 	}
 }
 
-func isTaskBlocked(ctx context.Context, at *restModel.APITask) (*bool, error) {
-	t, err := task.FindOneIdNewOrOld(*at.Id)
+func isTaskBlocked(ctx context.Context, at *restModel.APITask) (bool, error) {
+	i, err := at.ToService()
 	if err != nil {
-		return nil, ResourceNotFound.Send(ctx, err.Error())
+		return false, InternalServerError.Send(ctx, fmt.Sprintf("converting task '%s' to service", *at.Id))
 	}
-	if t == nil {
-		return nil, ResourceNotFound.Send(ctx, fmt.Sprintf("task %s not found", *at.Id))
+	t, ok := i.(*task.Task)
+	if !ok {
+		return false, InternalServerError.Send(ctx, fmt.Sprintf("converting APITask '%s' to Task", *at.Id))
 	}
 	isBlocked := t.Blocked()
-	return &isBlocked, nil
+	return isBlocked, nil
 }
 
-func isExecutionTask(ctx context.Context, at *restModel.APITask) (*bool, error) {
+func isExecutionTask(ctx context.Context, at *restModel.APITask) (bool, error) {
 	i, err := at.ToService()
 	if err != nil {
-		return nil, InternalServerError.Send(ctx, fmt.Sprintf("Error while converting task %s to service", *at.Id))
+		return false, InternalServerError.Send(ctx, fmt.Sprintf("converting task '%s' to service", *at.Id))
 	}
 	t, ok := i.(*task.Task)
 	if !ok {
-		return nil, InternalServerError.Send(ctx, fmt.Sprintf("Unable to convert APITask %s to Task", *at.Id))
+		return false, InternalServerError.Send(ctx, fmt.Sprintf("converting APITask '%s' to Task", *at.Id))
 	}
 	isExecutionTask := t.IsPartOfDisplay()
-
-	return &isExecutionTask, nil
+	return isExecutionTask, nil
 }
 
-func canRestartTask(ctx context.Context, at *restModel.APITask) (*bool, error) {
+func canRestartTask(ctx context.Context, at *restModel.APITask) (bool, error) {
 	taskBlocked, err := isTaskBlocked(ctx, at)
 	if err != nil {
-		return nil, err
+		return false, err
 	}
-	canRestart := !utility.StringSliceContains(evergreen.TaskUncompletedStatuses, *at.Status) || at.Aborted || (at.DisplayOnly && *taskBlocked)
-	isExecTask, err := isExecutionTask(ctx, at) // Cant restart execution tasks.
+	canRestart := !utility.StringSliceContains(evergreen.TaskUncompletedStatuses, *at.Status) || at.Aborted || (at.DisplayOnly && taskBlocked)
```

Comment:
Wanted to point out this existing code because it seems strange; apparently you are allowed to restart a task if it’s a Display Task and it’s blocked? Does that sound right? 🤔

Reply:
I think this makes sense since later tasks in a display task could be blocked on early tasks in the display task, in which case restarting the entire display task may unblock these.

However, I would recommend that you break this line into pieces for readability. Really what this function could be is:
if case1 { return false }
if case2 { return false }
if case3 { return false}
return true 

There's no reason to compute canRestart in the executionTask case for example; it makes sense to just check that first and return early.